usr/local/bin/install-um-extras

```diff
@@ -1,15 +1,119 @@
 #!/bin/bash
+# Install packages for the UM and UMDPs.
+# Allows a choice of GRIB_API and EcCodes, which are mutually exclusive.
 set -eu
 
+ubuntu_major=$(lsb_release -rs | cut -d. -f1)
+
+function usage {
+  # Print command line options
+  echo
+  echo 'Usage: um-setup [-e|--eccodes] [-g|--grib_api] [-v|--version <x.y>]'
+  echo
+  echo 'Options:'
+  echo '  -e, --eccodes        Install the EcCodes GRIB library'
+  echo '  -g, --grib           Install the GRIB_API GRIB library'
+  echo '  -h, --help           Show this help and exit'
+  echo '  -v, --version <x.y>  Install packages for the selected UM version'
+  echo
+  echo 'Provide at most one argument.'
+  echo 'Recommended usage is to indicate which UM version you wish to work with, e.g.'
+  echo '  um-setup -v 11.2'
+  echo 'For UM 11.2 or later, the EcCodes library for GRIB will be installed.'
+  echo 'For UM 11.1 or earlier, the GRIB_API library for GRIB will be installed.'
+  echo 'Alternatively, the --eccodes or --grib_api arguments may be provided to'
+  echo 'explicitly choose one library over another.'
+  echo 'These two libraries are mutually exclusive; only one may be installed ata time.'
+  echo
+  echo 'EcCodes is only available when running Ubuntu 18.04 or later, where it is'
+  echo 'the default. Earlier versions will default to the GRIB_API library.'
+  echo
+  echo 'Re-run this script with a different option to install the alternative library.'
+}
+
+function ereport {
+  # Print an error message, print usage, then exit (non-zero)
+  echo "${1:-Unknown error}"
+  usage
+  exit 1
+}
+
+function get_version {
+  # Turn an x.y version number into an integer
+  um_version=${1#vn}  # Remove leading vn, if present
+  um_major=$(echo $um_version | cut -d. -f1)
+  um_minor=$(echo $um_version | cut -d. -f2) # Ignores sub-releases (x.y.z)
+  um_version=$((10*um_major + um_minor))
+}
+
+function process_args {
+  # Parse and process the command line arguments
+  if [ $# -ge 1 ] ; then
+    case "$1" in
+      -e|--eccodes)
+          if [ $ubuntu_major -lt 18 ]; then
+            ereport "EcCodes installation requires Ubuntu 18 or later"
+          fi
+          grib_library=libeccodes-dev
+          ;;
+      -g|--grib_api|--grib-api)
+          grib_library=libgrib-api-dev
+          ;;
+      -h|--help)
+          usage
+          exit 0
+          ;;
+      -v|--version)
+          shift
+          get_version $1
+          if [ $# -eq 0 ] ; then
+            ereport "No UM version number provided"
+          fi
+          if [ $um_version -ge 112 ]; then
+            grib_library=libeccodes-dev
+          else
+            grib_library=libgrib-api-dev
+          fi
+          ;;
+       *) ereport "Unrecognised argument: $1"
+          ;;
+    esac
+    shift
+    # Only 1 option at a time is permitted:
```

Comment:
What happens if no arguments are provided? Presumably if you need one and only one we need a documentation update? (haven't checked to see if you've already done it, sorry)

Reply:
With no arguments it'll default to eccodes on Ubuntu 18, and grib-api on older releases.
https://github.com/paulcresswell/metomi-vms/blob/efe92f58c2645837113de0c4bdad044373ac21f2/usr/local/bin/install-um-extras#L93
That way it should continue working in the future without needing to add the argument each time.

It does need a UMDP update, but also for Shumlib, the change of supported virtual OS, etc. so that'll get done as part of the normal UMDP-X updates.